frontend/src/components/UserTables/LocationsTable.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useEffect } from 'react';
+import Form from '../UserForms/LocationsForm';
+import { Location } from '../../types';
+import './table.css';
+
+const Table = () => {
+  const [locations, setLocations] = useState<Location[]>([]);
+
+  const getExistingLocations = async () => {
+    const locationsData = await fetch('/locations')
+      .then((res) => res.json())
+      .then((data) => data.data);
+    setLocations(locationsData.map((location: any) => ({
+      id: location.id,
+      name: location.name,
+      address: location.address,
+    })));
+  };
+
+  useEffect(() => {
+    getExistingLocations();
+  }, []);
+
+  const addLocation = (newLocation: Location) => {
+    const requestOptions = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        name: newLocation.name,
+        address: newLocation.address,
+      }),
+    };
+    fetch('/locations', requestOptions).then((res) => {
+      if (res.status === 200) {
+        setLocations([...locations, newLocation]);
+      } else {
+        // TODO: alert
+      }
+    });
+  };
+
+  const deleteLocation = (locationId: string) => {
+    const requestOptions = {
+      method: 'DELETE',
+      headers: { 'Content-Type': 'application/json' },
+    };
+    fetch(`/locations/${locationId}`, requestOptions).then((res) => {
+      if (res.status === 200) {
+        setLocations(locations.filter((l) => l.id !== locationId));
+      } else {
+        // TODO: alert
```

Comment:
Maybe put a console.err here?

Reply:
I'm not sure if it's only on my end, but I receive an eslint warning `no-console`.

Reply:
If it's only giving a warning, then just ignore it. We can always remove it later.